R/check-annotation-values.R

```diff
@@ -321,12 +326,29 @@ check_type <- function(values, key, annotations, whitelist_values = NULL,
 #' can_coerce(2.5, "integer")
 #' }
 can_coerce <- function(values, class) {
+  ## If the values are already the correct class, then obviously the answer
+  ## should be yes
+  if (inherits(values, class)) {
+    return(TRUE)
+  }
+
+  ## Convert factors to strings
+  if (inherits(values, "factor")) {
+    values <- as.character(values)
+  }
+
   if (class == "character" &
     (inherits(values, "numeric") | inherits(values, "integer") |
       inherits(values, "logical"))) {
     return(TRUE)
   } else if (class == "numeric" & inherits(values, "integer")) {
     return(TRUE)
+  } else if (class == "logical") {
+    if (all(values %in% c("true", "True", "TRUE", "false", "False", "FALSE"))) {
+      return(TRUE)
+    } else {
+      return(FALSE)
+    }
   } else {
     return(FALSE)
   }
```

Comment:
Failing on `can_coerce(factor("foo"), "character")` because `class == "character"`, but "foo" is a character, not numeric, integer, or logical. Perhaps another check at the top for `inherits(values, "character")`.

Reply:
Actually I wonder if it would be better to add `inherits(values, "factor")` and remove the lines above where we convert factors to character. Not sure why I didn't think of that. Does that sound ok to you?

Reply:
That's a good idea.